galsim/phase_psf.py

```diff
@@ -0,0 +1,1442 @@
+                        galsim.Angle instance.  [default: 0*galsim.degrees]
+        @param compact  If true, then only return wavefront for illuminated pixels in a
+                        single-dimensional array congruent with array[aper.illuminated].  Otherwise,
+                        return wavefront as a 2d array for the full Aperture pupil plane.
+                        [default: True]
+        @returns        Wavefront lag or lead in nanometers over aperture.
+        """
+        return np.sum(layer.wavefront(aper, theta_x, theta_y, compact) for layer in self)
+
+    def makePSF(self, lam, **kwargs):
+        """Compute one PSF or multiple PSFs from the current PhaseScreenList, depending on the type
+        of (`theta_x`, `theta_y`) or `theta`.  If (`theta_x`, `theta_y`) or `theta` are iterable,
+        then return a list of PSFs at the specified field angles.  If `theta_x` and `theta_y` are
+        scalars or `theta` is a single tuple, then return a single PSF at the specified field angle.
+
+        Note that this method advances each PhaseScreen in the PhaseScreenList, so consecutive calls
+        with the same arguments will generally return different PSFs.  Use PhaseScreenList.reset()
+        to reset the time to t=0.  See the galsim.PhaseScreenPSF docstring for more details.
+
+        @param lam               Wavelength in nanometers at which to compute PSF.
+        @param exptime           Time in seconds overwhich to accumulate evolving instantaneous PSF.
+                                 [Default: 0.0]
+        @param flux              Flux of output PSF [Default: 1.0]
+        @param theta_x           x-component of field angle at which to evaluate phase screens and
+                                 resulting PSF.  Either a single galsim.Angle, or an iterable of
+                                 galsim.Angles.  [Default: 0.0*galsim.arcmin]
+        @param theta_y           y-component of field angle at which to evaluate phase screens and
+                                 resulting PSF.  Either a single galsim.Angle, or an iterable of
+                                 galsim.Angles.  [Default: 0.0*galsim.arcmin]
+        @param theta             Alternative field angle specification.  Single tuple or iterable of
+                                 tuples (theta_x, theta_y).
+        @param scale_unit        Units to use for the sky coordinates of the output profile.
+                                 [Default: galsim.arcsec]
+        @param interpolant       Either an Interpolant instance or a string indicating which
+                                 interpolant should be used.  Options are 'nearest', 'sinc',
+                                 'linear', 'cubic', 'quintic', or 'lanczosN' where N should be the
+                                 integer order to use. [default: galsim.Quintic()]
+        @param aper              Aperture to use to compute PSF(s).
```

Comment:
[default: None] here.  Also for `theta`.  And some of these default specifications have a capital `Default`.  Our style has been to use lowercase for this.